pyiron_base/jobs/job/core.py

```diff
@@ -429,6 +431,48 @@ def remove(self, _protect_childs=True):
         # After all children are deleted, remove the job itself.
         self.remove_child()
 
+    def kill(self, enable=True):
+        """
+        Function that kills a job via queueing system commands (e.g. scancel/qdel in terminal)
+        It also updates the status of the job to "aborted" in the job database.
+
+        enable (bool): Attempt a job-kill regardless of status when set to True.
+                       Set to False if you don't know what you are doing (unlikely).
+
+        Use case: A python-side equivalent method to call "scancel JOBID" (slurm) or "qdel JOBID" (PBSPro/torque) via pyiron
+
+        This command is "dumb", it will attempt to kill anything that it is called on, regardless of job-status.
+        """
+
+        def base_kill():
+            if "server" in self.project_hdf5.list_nodes():
+                server_hdf_dict = self.project_hdf5["server"]
+                if (
+                    "qid" in server_hdf_dict.keys()
+                    and server_hdf_dict["qid"] is not None
+                ):
+                    self.project.queue_delete_job(server_hdf_dict["qid"])
+                    self._status = "aborted"
```

Comment:
Please use `self.status` then the database is also updated.

Reply:
https://github.com/pyiron/pyiron_base/blob/master/pyiron_base/jobs/job/extension/jobstatus.py#L196

Reply:
Since we're operating on `JobCore` that is not automatically called.

Reply:
> Since we're operating on `JobCore` that is not automatically called.

Oh, ok now I see where the trouble is coming from. But then would not it be easier to implement this action on the job level rather than the jobcore level?

Reply:
I am just seeing that `remove_child` is inherited from the jobcore level (right above this).

`remove_child is part of the `remove` functionality in core.

`remove_child` now calls the kill functionality as part of it's operation (as per your suggestion to avoid duplicated code)

To me, it is weird to have one inherit from one module and have another function right below it have the opposite behaviour coded in the module and inherited by the `core` module. This doesn't make much sense especially since now it is part of core functionality in the `remove_child` method.

The old kill method also derived from core - by implementing this in the generic level this represents a reversal in architecture.

Design wise, it would be preferable to me to have all the core functionality in one place as opposed to having interdependencies between packages. (even if it exists currently)

I feel that is important for code maintainability and understandability overall.